learning-to-code/arrays/README.md

```diff
@@ -48,40 +51,63 @@ We can utilize the syntax for both of the above to initialize a new Array _with
-![numbers array](./images/array-1.png)
 
 We can do something similar when creating an array using `Array.new`. When passing parameters into `new`, the first parameter will be the size of the desired array. The second parameter will be the default value for all element within the array.
 
 ```ruby
 empty_array = Array.new(3)
 empty_array # => [nil, nil, nil]
 ```
-![empty_array](./images/array-2.png)
 
 ```ruby
 default_array = Array.new(3, "def")
 default_array # => ["def", "def", "def"]
 ```
-![default_array](./images/array-3.png)
 
 ### Indices
-Arrays are integer-indexed starting at __zero__. All counting in computer science [begins with zero](http://skillcrush.com/2013/01/17/why-programmers-start-counting-at-zero/). This means that **each item** in the array corresponds to an **integer value**, and that integer is used to access an object within the `Array`. The first object is assigned 0 and increments up from there.
+Arrays are integer-indexed starting at __zero__. All counting in computer science [begins with zero](http://skillcrush.com/2013/01/17/why-programmers-start-counting-at-zero/). This means that **each item** in the array corresponds to an **integer value**, and that integer is used to access an object within the `Array`. The first object is at index 0, the second object is at index 1 and so on.
 
 ```ruby
 numbers = [1, 10, 33, 50, 2, 7]
-numbers[0] # => 1
-numbers[1] # => 10
-numbers[2] # => 33
-numbers[3] # => 50
-numbers[4] # => 2
-numbers[5] # => 7
+puts numbers[0] # => 1
+puts numbers[1] # => 10
+puts numbers[2] # => 33
+puts numbers[3] # => 50
+puts numbers[4] # => 2
+puts numbers[5] # => 7
 ```
-![my_array](./images/array-4.png)
+| Value |  1  |  10 |  33 |  50 | 2   |  7  |
+| :---- | :-- | :-- | :-- | :-- | :-- | :-- |
+| Index | [0] | [1] | [2] | [3] | [4] | [5] |
 
-### Data Types
-Arrays can store all sorts of data. Usually, it makes the most sense to have a single data type in an array, but in Ruby, it is not a requirement.
 
-This means that arrays can store data types of `Integer`, `String`, and `Float`!
+```ruby
+numbers = [1, 2, 3, 4]
+puts numbers[2] # => 3
+```
+| Value |  1  |  2 |  3 |  4 |
+| :---- | :-- | :-- | :-- | :-- |
+| Index | [0] | [1] | [2] | [3] |
+
+```ruby
+empty_array = Array.new(3)
+puts empty_array # => [nil, nil, nil]
+```
+| Value |  nil  |  nil |  nil |
+| :---- | :-- | :-- | :-- |
+| Index | [0] | [1] | [2] |
+
+```ruby
+default_array = Array.new(3, "def")
+puts default_array # => ["def", "def", "def"]
+```
+| Value | "def" |  "def" |  "def" |
+| :---- | :-- | :-- | :-- |
+| Index | [0] | [1] | [2] |
+
+
+### Data Types
+Arrays can store all sorts of data. In most programming languages, arrays are _homogeneous_ data structures. This means that in most programming lanaguages, the same type of data is stored in each element of the array. e.g. an array of integers, an array of strings etc. However, in Ruby, it is not a requirement for all elements in the array to be of the same type.
 
```

Comment:
I hear you. We may not need to get into the details during JumpStart. I will reword it to focus only on how Ruby does it in the JumpStart curriculum.

*Thoughts for offline discussion*: Keeping in mind the idea of learning how to learn a new language (since a lot of the students will have to learn Java, C++ or C# during internship), and to make it easier for the students to grasp data structure concepts later, my motive here was to call out that Ruby's Array class is a special, language specific construct. 

1. The Array class in Ruby does not follow the same rules as an Array data structure. [Array data structure](https://www.geeksforgeeks.org/overview-of-data-structures-set-1-linear-data-structures/): data structure used to store homogeneous elements at contiguous locations. Arrays are also static data structures, unlike Ruby's Array class which can dynamically grow and shrink in size.
2. There are other C like languages (C, C++, C#, Java), where arrays of primitive data types always mean homogeneous elements at contiguous location. Some of the higher level languages like C# and Java provide other classes like ArrayList, which provide more flexibility like Ruby's Array class provides.
This gets covered in the second CS Fundamental's class. Are there other places in the curriculum where we could underscore this again?

Reply:
It's definitely an idea that we want them to get to eventually - I think early in the CS fun curriculum is a good spot for this.